### ISSUE TITLE
Question on CMLM Implementation

### ISSUE THREAD

**Issue by julianalverio** [BytedProtein/ByProt#2]
Hi,

I have a question about your implementation of the CMLM. Which model did you use to select the least certain tokens to mask and then iteratively infill them? Do you look at the final output of your LM head to determine which logits are most uncertain? Or do you use a frozen ESM-1b network and head to determine this? Secondly, do you then provide the masked version of the output sequence to the language model and train a separate LM head for masked token prediction? Or do you feed this masked sequence back into the structure encoder and perform a full pass through the LMDesign network to predict the masked token? @zhengzx-nlp 

Thanks!

**Comment by zhengzx-nlp** [comment]
Hi! I found I didn't reply to you and I sincerely apologize! Please check our implementation of sampling in https://github.com/BytedProtein/ByProt/blob/main/src/byprot/models/fixedbb/generator.py
